### PR TITLE
[WIP] Improve Styling

### PR DESCRIPTION
**Pull request by gunosk129**

### CODE REVIEW

**Comment by gunosk129** [comment]
@hiattp When we first implement it, Oo and I thought I might be a flash message because it is the same design with the referral page so we make it be a flash message. So the red banner on landing page shouldn't be flash message, but on referral page, it's flash message am I right?

**Comment by hiattp** [comment]
Neither is a flash message, they are just parts of the designs of each of
those pages

[image: Lockstep Logo] <https://www.locksteplabs.com/> Paul Hiatt
M: +1 (908) 442-8875
W: locksteplabs.com <https://www.locksteplabs.com>
<https://www.linkedin.com/in/paulhiatt>  <https://github.com/hiattp>
<http://stackoverflow.com/users/949328/hiattp>
<https://twitter.com/paulrugelhiatt>

File: app/assets/javascripts/landing_page.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+$(document).ready(() => {
+  if ($('.landing-page').length > 0) {
+
+    $('#subscription_email').on('keypress', () => {
+      const submitButton = $('#submit-email');
+
+      if (this.value !== '') {
```
Comment (citizen428) [inline app/assets/javascripts/landing_page.js]: It's generally preferred to have positive conditionals, in this case:

```javascript
if (this.value === '') {
  submitButton.prop('disabled', true);
} else {
  submitButton.prop('disabled', false);
}

File: app/assets/javascripts/referral_page.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  return copySuccess;
+}
+
+$(document).ready(() => {
+  if ($('.referral-page').length > 0) {
+    const input = $('.referral-input');
+    input.focus();
+    input.select();
+
+    input.on('click', () => {
```
Comment (citizen428) [inline app/assets/javascripts/referral_page.js]: No need for the curly braces for a single statement arrow function:

```javascript
  input.on('click', () => copySelectionText())
```

File: app/models/subscription.rb [hunk 3]
```diff
@@ -9,7 +9,8 @@ class Subscription < ApplicationRecord
              foreign_key: :referrer_id, counter_cache: :invited_count
   has_many :invitees, class_name: 'Subscription', foreign_key: :referrer_id
 
-  validates :email, presence: true, uniqueness: true
+  validates :email, format: { with: /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\Z/i },
```
Comment (citizen428) [inline app/models/subscription.rb]: You may just want to use `Devise.email_regexp`:

```ruby
Devise.email_regexp
#=> /\A[^@\s]+@[^@\s]+\z/
```

File: spec/features/landing_page_spec.rb [hunk 4]
```diff
@@ -14,15 +14,21 @@
     visit '/'
 
     fill_in 'subscription_email', with: 'user1@email.com'
-    click_button 'Request Invite'
-    expect(page).to have_content 'Thanks for signing up'
+    find('#submit-email').click
+    expect(page).to have_content 'THANKs FOR SIGNING UP'
```
Comment (citizen428) [inline spec/features/landing_page_spec.rb]: The `s` should probably not be lowercase.

**Comment by gunosk129** [comment]
@hiattp @citizen428 I finished all the styling stuff except hi-res image, please review it.

**Comment by hiattp** [comment]
@gunosk129 can you correct conflicts and merge to development so the client can review on staging?

**Pull request merged**